src/components/TopNav/NavMenu.js

```diff
@@ -7,19 +7,44 @@ import { Link } from 'react-router-dom'
 import NavigationMenu from 'material-ui/svg-icons/navigation/menu'
 import IconButton from 'material-ui/IconButton'
 import * as auth from 'services/authNService'
+import { _ } from 'underscore'
 
-export const NavMenu = ({ dispatch }) =>
-  <IconMenu
+const Tickets = () => {
+  var persistedTickets  = JSON.parse(localStorage['persist:ticket'])
+  var currentTicketId   = window.location.pathname.match(/(\d+)/)[1]
 
+  var idIsNumericalAndNotCurrentTicket  = (id) => /\d+/.test(id) && id != currentTicketId
+  var addCurrentPageToHistory           = () => { history.push(this.to) }
```

Comment:
I think the Link component does this automatically; if it doesn't then we have other changes to make. We only add onClick for some of them to have effects other than changing the current page.

Reply:
The Link component does not handle this automatically.

A Link created without this onClick will generate the correct link, but when I click through, I get an "Unexpected Failure When Attempting to Display Incident" message. This error is raise in Ticket in the code sample below:

``` javascript
    if (!incident || !incident.primaryTicket || !ticket || incident.error) {
      return UnexpectedFailureToLoadIncident()
    }
```

When a link to an incident is loaded from the navMenu, this code is triggered, with undefined for incident and an object for ticket.

Reply:
https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/modules/Link.js

Looking at the source code for link, it definitely should be doing history.push(to)